base/tuple.jl

```diff
@@ -310,3 +310,23 @@ any(x::Tuple{}) = false
 any(x::Tuple{Bool}) = x[1]
 any(x::Tuple{Bool, Bool}) = x[1]|x[2]
 any(x::Tuple{Bool, Bool, Bool}) = x[1]|x[2]|x[3]
+
+## tuple-type introspection as array/iterator ##
+# ... we can't define iteratorsize/iteratoreltype, however, because
+#     typeof(Tuple) is simply DataType
+_iteratorsize(::Type{T}) where {T<:Tuple} = (@_pure_meta; isvatuple(T) ? IsInfinite() : HasLength())
+_length(::HasLength, T::Type{<:Tuple}) = length(T.types)
+_length(::IsInfinite, T::Type{<:Tuple}) = typemax(Int)
```

Comment:
Should it throw an error instead?   (I could just delete this line and it would throw a `MethodError` in that case.)